extensions/amp-compare-slider/0.1/amp-compare-slider.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+
+export class AmpCompareSlider extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.myText_ = 'hello world';
+
+    /** @private {!Element} */
+    this.container_ = this.win.document.createElement('div');
```

Comment:
Please delay document API until `#buildCallback`. Only initiate `this.container_` in `#constructor`, and assign its value in `#buildCallback`